src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -97,21 +113,99 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
     toggle: togglePersonalisedSchedule,
   } = useShowHide(false);
 
+  const liveAndFutureEvents = useMemo(
+    () =>
+      [
+        ...relatedVenueEvents,
+        ...relatedVenueEvents,
+        ...relatedVenueEvents,
+        ...relatedVenueEvents,
+        ...relatedVenueEvents,
+      ]
+        .filter(isEventLiveOrFuture)
+        .map(
+          prepareForSchedule({
+            usersEvents: userEventIds,
+          })
+        ),
+    [relatedVenueEvents, userEventIds]
+  );
+  const hasSavedEvents = !!liveAndFutureEvents.filter((event) => event.isSaved)
+    .length;
+
+  const isShowPersonalDownloadBtn = hasSavedEvents && showPersonalisedSchedule;
+  const liveEventsMinimalStartValue = Math.min(
+    ...liveAndFutureEvents.map((event) => event.start_utc_seconds)
+  );
+
+  const firstLiveEvent = liveAndFutureEvents.find(
+    (event) => event.start_utc_seconds === liveEventsMinimalStartValue
+  );
+
+  const minDate = useMemo(
```

Comment:
the name of variable is a bit confusing. Let's rename it to minDateUtcSeconds.
and then we can create a separate 
```js
const minDate = useMemo(() => fromUnixTime(minDateUtcSeconds), [minDateUtcSeconds]);
```

actually, you can check if we ever need utcSeconds. Maybe it's more comfortable to store date object instead?

Reply:
renamed that variable in 12ae094
we still need the utcseconds value for some functions down below tho